lib/inc/hocon/config_object.hpp

```diff
@@ -27,6 +28,9 @@ namespace hocon {
         virtual shared_object with_value(path raw_path, shared_value value) const = 0;
         virtual shared_object with_value(std::string key, shared_value value) const = 0;
 
+        shared_value with_fallback(std::shared_ptr<const config_mergeable> mergeable);
+        static shared_origin merge_origins(std::vector<shared_value> stack);
```

Comment:
mergeOrigins is part of the internal implementation, so it should go in the `protected` block.